src/sage/combinat/k_regular_sequence.py

```diff
@@ -918,6 +979,367 @@ def _n_to_index_(self, n):
+                               [18/5 27/5],
+                            2: [ 6  0]
+                               [24  2]},
+             (1, 1))
+            sage: G == S.partial_sums(include_n=True)
+            True
+
+        ::
+
+            sage: Seq2.guess(s, max_exponent=1)
+            Traceback (most recent call last):
+            ...
+            RuntimeError: aborting as exponents would be larger than max_exponent=1
+        """
+        import logging
+        logger = logging.getLogger(__name__)
+
+        from sage.arith.srange import srange, xsrange
+        from sage.matrix.constructor import Matrix
+        from sage.misc.mrange import cantor_product
+        from sage.modules.free_module_element import vector
+
+        k = self.k
+        domain = self.coefficient_ring()
+        if sequence is None:
+            mu = [[] for _ in srange(k)]
+            seq = lambda m: vector([])
+        else:
+            mu = [M.rows() for M in sequence.mu]
+            seq = lambda m: sequence.left * sequence._mu_of_word_(
+                self._n_to_index_(m))
+            logger.info('including %s', sequence)
+
+        zero = domain(0)
+        one = domain(1)
+
+        # A `line` will be a triple `(t, r, s)` corresponding to an entry
+        # `k**t * m + r` belonging to `M_s`
+        # TODO: what is `M_s`?
+
+        # The elements of `lines` will be correspond to the current components
+        # of the left vector valued sequence described in the algorithm section
+        # of the docstring.
+
+        def values(m, lines):
+            """
+                Return current (as defined by ``lines``) left vector valued
+                sequence for argument ``m``.
+            """
+            return tuple(seq(m)) + tuple(f(k**t_R * m + r_R) for t_R, r_R in lines)
+
+        @cached_function(key=lambda lines: len(lines))  # we assume that existing lines are not changed (we allow appending of new lines)
+        def some_inverse_U_matrix(lines):
+            r"""
+                Find an invertible `d \times d` times submatrix of the matrix
+                ``A`` described in the algorithm section of the docstring.
+
+                The output is the inverse of the invertible submatrix and
+                the corresponding list of column indices (i.e., arguments to
+                the current left vector valued sequence).
+            """
+            d = len(seq(0)) + len(lines)
+
+            for m_indices in cantor_product(xsrange(n_max), repeat=d, min_slope=1):
+                # Iterate over all increasing lists of length d consisting
+                # of non-negative integers less than `n_max`.
+
+                U = Matrix(domain, d, d, [values(m, lines) for m in m_indices]).transpose()
+                try:
+                    return U.inverse(), m_indices
+                except ZeroDivisionError:
+                    pass
+            else:
+                raise RuntimeError('no inverse submatrix found')
```

Comment:
```suggestion
                raise RuntimeError('no invertible submatrix found')
```
would perhaps be more correct.